### DISCUSSION TITLE
Reactive formState

### DISCUSSION THREAD

**Discussion by Trollyal** [react-hook-form/react-hook-form#8029] [Q&A] [answered]
Hi, i'd like to use the formState and it's values like `isValid` to disable buttons etc. The problem is that the `formState` is updated in batch and therefore is out of sync with reality (`isValid` is false, but when i hit submit, it triggers the `isValid callback`). Is there way for the `formState` to be updated on any change and not in batch?

**Comment by Trollyal** [comment]
The use case:
https://stackblitz.com/edit/react-ts-2mlfcs?file=index.tsx

The dynamic validation works, but the results are hindered by the lazy batch  update of formState which disallows me to use the validity to disable button.

The form is is valid if either or both fields are filled. You can see, how the `isValid` is not updated correctly in comparison to what you get in console, when you hit submit. Try it multiple times, if you only fill the right fields it tends to work, not the other way around.

**Comment by Trollyal** [comment, marked as answer]
My bad, was looking in a wrong direction. The answer was wrong usage of `deps` which i thought were `dependencies` but they are `dependants` so i was putting them in wrong register.

**Reply by Moshyfawn** [reply]
That's a nice insight, thx, @Trollyal!

@bluebill1049 what are your thoughts on swapping `deps` to `dependants` or `dependentFields` in the next major release? I know it's a longer name, but I usually see `deps` used in the ctx of `dependencies` and it can help avoid this sort of confusion.

**Reply by bluebill1049** [reply]
sounds good to me @Moshyfawn let's change it next version.

**Reply by bluebill1049** [reply]
perhaps it should just called `trigger` `register('test', {trigger: []})`

**Reply by Moshyfawn** [reply]
That actually makes sense to me!
